### ISSUE TITLE
docker: Potential issues with bootstrapping when running node in docker on macos

### ISSUE THREAD

**Issue by vlad2095** [celestiaorg/celestia-node#3533]
### Celestia Node version

0308aea76857ff27484946fce99004ebf10a3cb8 / v0.13.7

### OS

ghcr.io/celestiaorg/celestia-node container on MacOS M1.

### Install tools

docker/podman

### Steps to reproduce it

celestia light init --p2p.network celestia --core.ip rpc.celestia.pops.one
celestia light start --p2p.network celestia --core.ip rpc.celestia.pops.one

### Expected result

successfully running light node

### Actual result

Error: node: failed to start: header/p2p: failed to open a new stream: failed to dial: failed to dial 12D3KooWQpuTFELgsUypqp9N4a1rKBccmrmQVY8Em9yhqppTJcXf: all dials failed
  * [/ip4/163.172.132.189/tcp/2121] dial backoff

### Relevant log output

```Shell
2024-06-18T06:08:16.254Z        INFO    node    nodebuilder/module.go:26        Accessing keyring...
2024-06-18T06:08:16.275Z        INFO    badger4 v4@v4.2.1-0.20240106094458-1c417aa3799c/levels.go:171   All 1 tables opened in 1ms
2024-06-18T06:08:16.276Z        INFO    badger4 v4@v4.2.1-0.20240106094458-1c417aa3799c/discard.go:66   Discard stats nextEmptySlot: 1
2024-06-18T06:08:16.276Z        INFO    badger4 v4@v4.2.1-0.20240106094458-1c417aa3799c/db.go:368       Set nextTxnTs to 143
2024-06-18T06:08:17.150Z        INFO    pidstore        pidstore/pidstore.go:67 Loaded peers from disk  {"amount": 0}
2024-06-18T06:08:17.150Z        INFO    module/header   header/config.go:66     No trusted peers in config, initializing with default bootstrappers as trusted peers
2024-06-18T06:08:17.170Z        INFO    header/p2p      p2p/subscriber.go:81    joining topic   {"topic ID": "/celestia/header-sub/v0.0.1"}
2024-06-18T06:08:17.170Z        INFO    header/p2p      p2p/exchange.go:99      client: starting client {"protocol ID": "/celestia/header-ex/v0.0.3"}
2024-06-18T06:08:17.170Z        INFO    pidstore        pidstore/pidstore.go:67 Loaded peers from disk  {"amount": 0}
2024-06-18T06:08:17.468Z        INFO    pidstore        pidstore/pidstore.go:84 Persisted peers successfully    {"amount": 0}
2024-06-18T06:08:17.468Z        WARN    header/p2p      p2p/subscriber.go:92    unregistering validator: no validator for topic /celestia/header-sub/v0.0.1
2024-06-18T06:08:17.468Z        INFO    basichost       basic/natmgr.go:112     DiscoverNAT error:no NAT found
2024-06-18T06:08:17.469Z        INFO    badger4 v4@v4.2.1-0.20240106094458-1c417aa3799c/db.go:546       Lifetime L0 stalled for: 0s

2024-06-18T06:08:17.476Z        INFO    badger4 v4@v4.2.1-0.20240106094458-1c417aa3799c/db.go:625       
Level 0 [ ]: NumTables: 00. Size: 0 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 16 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 01. Size: 3.5 KiB of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
Error: node: failed to start: header/p2p: failed to open a new stream: failed to dial: failed to dial 12D3KooWQpuTFELgsUypqp9N4a1rKBccmrmQVY8Em9yhqppTJcXf: all dials failed
  * [/ip4/163.172.132.189/tcp/2121] dial backoff
```

### Notes

While I tried to run light node in any of p2p networks, it fails to connect via p2p

However, it seems like network-related issue, 'cause when I used mobile internet provider, it successfully started.
I remember having similar issue in one of my projects, where we solved it by increasing p2p timeout a bit.

Is it possible to configure celestia with increased p2p timeout or solve the issue in some other way?

**Comment by renaynay** [comment]
@vlad2095 what network are you trying to connect to? if it's mainnet, you do not have to pass the `--p2p.network` flag, or if you want to, you can pass `--p2p.network mainnet`

**Comment by vlad2095** [comment]
> @vlad2095 what network are you trying to connect to? if it's mainnet, you do not have to pass the `--p2p.network` flag, or if you want to, you can pass `--p2p.network mainnet`

Hi @renaynay I have the same results

Is there a way to increase p2p timeout?

<img width="1519" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/1c83951b-5adf-4cb4-8dc5-4049688beb62">

<img width="1488" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/3cc46a06-c2c5-4ace-b708-5e3fdfca2c5f">

**Comment by vlad2095** [comment]
Attempt to run manually from this repo was a success:

```
make build
./build/celestia light init --p2p.network celestia --core.ip rpc.celestia.pops.one
./build/celestia light start --p2p.network celestia --core.ip rpc.celestia.pops.one
```

Output:

```
Started celestia DA node 
node version:   v0.14.0-14-g1fa5aa6a
node type:      light
network:        celestia

/_____/  /_____/  /_____/  /_____/  /_____/ 

The p2p host is listening on:
*  /ip4/79.110.128.121/tcp/36799/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
2024-06-27T15:33:08.031+0300    INFO    das     das/worker.go:94        finished sampling headers       {"type": "catchup", "from": 1, "to": 1, "errors": 0, "# of headers skipped as outside of sampling window": 1, "finished (s)": 0.000020708}
*  /ip4/79.110.128.121/udp/36799/quic-v1/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip4/79.110.128.121/udp/36799/quic-v1/webtransport/certhash/uEiBJc3-yrgTAxE_cxd25ZXMRu4x1GN-A0ZxudceRSoTmew/certhash/uEiDsXeQEqY3k3fWoFW3_JLwyV2TmK-R-J4z9zHN21eNSaA/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip4/127.0.0.1/udp/2121/quic-v1/webtransport/certhash/uEiBJc3-yrgTAxE_cxd25ZXMRu4x1GN-A0ZxudceRSoTmew/certhash/uEiDsXeQEqY3k3fWoFW3_JLwyV2TmK-R-J4z9zHN21eNSaA/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip4/192.168.50.254/tcp/2121/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip4/192.168.50.254/udp/2121/quic-v1/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip4/192.168.50.254/udp/2121/quic-v1/webtransport/certhash/uEiBJc3-yrgTAxE_cxd25ZXMRu4x1GN-A0ZxudceRSoTmew/certhash/uEiDsXeQEqY3k3fWoFW3_JLwyV2TmK-R-J4z9zHN21eNSaA/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip6/::1/tcp/2121/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip6/::1/udp/2121/quic-v1/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD
*  /ip6/::1/udp/2121/quic-v1/webtransport/certhash/uEiBJc3-yrgTAxE_cxd25ZXMRu4x1GN-A0ZxudceRSoTmew/certhash/uEiDsXeQEqY3k3fWoFW3_JLwyV2TmK-R-J4z9zHN21eNSaA/p2p/12D3KooWJasWutWWCjKxiuJSg3Zg47VnvGrU8rNgmJEhyARAvVyD

2024-06-27T15:33:12.495+0300    INFO    share/discovery discovery/discovery.go:325      discovered wanted peers {"topic": "full", "amount": 5}
```

**Comment by renaynay** [comment]
hey @vlad2095 my apologies, `celestia` is actually a valid input for `--p2p.network`

Interesting -- there might be some issues with reachability running the node in docker on Mac. We will investigate, thanks.

**Comment by vlad2095** [comment]
@renaynay not exactly related to Mac.
I thought that too, but when I switched internet connection to mobile 4G I was able to connect.
Also my friend from Lisbon didn't have such issues in the first place while running the same way in Docker on M1

**Comment by renaynay** [comment]
Thanks for the info @vlad2095

**Comment by vlad2095** [comment]
@renaynay I was just pulled celestia-node repo and played a bit with.
While with default config it successfully run on my Mac 👍  
But when I decreased it by adding libp2p timeout config below 
[nodebuilder/p2p/host.go#L85](https://github.com/celestiaorg/celestia-node/blob/86f4b2aa1dd54a390dd213d5011b48ab51aa7321/nodebuilder/p2p/host.go#L85)

```
libp2p.WithDialTimeout(time.Microsecond * 100)
```
I experience same result as I did inside Docker

<img width="1402" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/e9335788-369c-48f2-bc3d-e9ac02e87db7">

While increasing it to 500 millisecond fixed the issue

<img width="1193" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/5c7b2b17-0c6c-400d-b7d4-4017121cff41">


--
So I guess Docker introduce some delays.
Is it possible to make this parameter configurable, with a flag?

**Comment by Wondertan** [comment]
@vlad2095, that must be a flake. The default setting for this timeout is [15 seconds](https://github.com/libp2p/go-libp2p/blob/6c178d4edf9a0fc50cd4783550af904ec9ae6e52/p2p/net/swarm/swarm.go#L28), so setting 500ms decreases the timeout.

**Comment by Wondertan** [comment]
(I don't have any good ideas on what could cause this issue for you. The only fact that 4G works suggests that this is unlikely related to Docker)

**Comment by Wondertan** [comment]
@vlad2095, another thing you could try is increasing the StartupTimeout in the node config

**Comment by vlad2095** [comment]
@Wondertan how would I increase StartupTimeout?

Yeah and 4G in my region actually is slower than cabel, so maybe it's a more "low level" network issue.
We tested in different location on MacOS with docker, and getting different results, like it works in Portugal but fails in Poland and Ukraine.

**Comment by Wondertan** [comment]
> how would I increase StartupTimeout?

In the config `.celestia-light/config.toml`

**Comment by vlad2095** [comment]
unfortunately, doesn't work
<img width="1391" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/7541d79f-4a97-41ac-9b42-58a82cc4e029">

**Comment by Wondertan** [comment]
Can you ping those IPs? At least to see that they are accessible from your network on ICMP level

**Comment by vlad2095** [comment]
Looks like they are
<img width="507" alt="image" src="https://github.com/celestiaorg/celestia-node/assets/29329498/118edefd-20f3-49d3-bfd4-64ebc0c710e3">

**Comment by smuu** [comment]
Hello @vlad2095,

If this issue only happens when running the node in a docker container, please try the following:

- Make sure your Macbook has the latest updates
- Make sure Docker Desktop has the latest updates
- Restart Docker Desktop
- Factory reset Docker Desktop
- Restart your Macbook

Please let me know if the issue still persists after trying these things.

**Comment by ramin** [comment]
@vlad2095 can we consider this resolved? Please close if so

**Comment by glihm** [comment]
> [@vlad2095](https://github.com/vlad2095) can we consider this resolved? Please close if so

Gm, I'm recently using Celestia and experiencing the same issue. I'm also on MacOS.
Trying to isolate the behavior, because in some moment, everything run just fine.

Here's an output on my end before the node stops:
```
Initializing Celestia Node with command:
celestia light init --p2p.network mocha --rpc.addr=0.0.0.0 --gateway.addr=0.0.0.0
2025-03-18T00:56:51.979Z        WARN    module/gateway  gateway/flags.go:47     custom address or port provided without enabling gateway, setting config values
2025-03-18T00:56:51.979Z        INFO    node    nodebuilder/init.go:31  Initializing Light Node Store over '/home/celestia'
2025-03-18T00:56:51.979Z        INFO    node    nodebuilder/init.go:64  Saved config    {"path": "/home/celestia/config.toml"}
2025-03-18T00:56:51.979Z        INFO    node    nodebuilder/init.go:66  Accessing keyring...
2025-03-18T00:56:51.981Z        WARN    node    nodebuilder/init.go:196 Detected plaintext keyring backend. For elevated security properties, consider using the `file` keyring backend.
2025-03-18T00:56:51.983Z        INFO    node    nodebuilder/init.go:73  Node Store initialized


Starting Celestia Node with command:
celestia light start --core.ip rpc-mocha.pops.one --core.port 9090 --p2p.network mocha

2025-03-18T00:56:52.009Z        INFO    node    nodebuilder/module.go:27        Accessing keyring...
2025-03-18T00:56:52.012Z        INFO    badger4 v4@v4.3.0/levels.go:171 All 2 tables opened in 0s

2025-03-18T00:56:52.013Z        INFO    badger4 v4@v4.3.0/discard.go:66 Discard stats nextEmptySlot: 6

2025-03-18T00:56:52.013Z        INFO    badger4 v4@v4.3.0/db.go:369     Set nextTxnTs to 11944
2025-03-18T00:56:52.046Z        INFO    module/p2p      p2p/tls.go:54   using a default ws transport
2025/03/18 00:56:52 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.
2025-03-18T00:56:52.051Z        INFO    pidstore        pidstore/pidstore.go:67 Loaded peers from disk  {"amount": 88}
2025-03-18T00:56:52.051Z        INFO    module/header   header/config.go:66     No trusted peers in config, initializing with default bootstrappers as trusted peers
2025-03-18T00:56:52.056Z        INFO    header/p2p      p2p/exchange.go:99      client: starting client {"protocol ID": "/mocha-4/header-ex/v0.0.3"}
2025-03-18T00:56:52.056Z        INFO    pidstore        pidstore/pidstore.go:67 Loaded peers from disk  {"amount": 88}
2025-03-18T00:56:52.056Z        INFO    header/store    store/store.go:193      loaded head     {"height": 13313, "hash": "06EDFAEEA2DB489D8682045D1A51973C44C77A4922A9400BF7F65D7E552E3997"}
2025-03-18T00:56:52.056Z        INFO    share/discovery discovery/discovery.go:131      started discovery       {"topic": "/full/v0.1.0"}
2025-03-18T00:56:52.056Z        INFO    share/discovery discovery/discovery.go:131      started discovery       {"topic": "/archival/v0.1.0"}
2025-03-18T00:56:52.057Z        INFO    fraudserv       fraudserv/helpers.go:29 joining topic   {"id": "/mocha-4/fraud-sub/badencodingv0.1/v0.0.1"}
2025-03-18T00:56:52.057Z        INFO    fraudserv       fraudserv/service.go:114        starting fraud proof service    {"protocol ID": "/mocha-4/fraud/v0.0.1"}
2025-03-18T00:56:52.502Z        INFO    canonical-log   swarm/swarm_dial.go:620 CANONICAL_PEER_STATUS: peer=12D3KooWLBEpFH54Q8ofQ5dLjFwXCom9BqUKJyP2d7A9oCXg5Kvo addr=/ip4/89.58.17.242/udp/2121/quic-v1 sample_rate=100 connection_status="established" dir="outbound"
2025-03-18T00:56:52.505Z        INFO    canonical-log   swarm/swarm_dial.go:620 CANONICAL_PEER_STATUS: peer=12D3KooWFMhDMU12zZ77scNG9UYv4N2jQa12p9Fm9XrQAuqswtPq addr=/ip4/85.10.196.25/udp/2121/quic-v1 sample_rate=100 connection_status="established" dir="outbound"
2025-03-18T00:56:57.051Z        INFO    basichost       basic/natmgr.go:112     DiscoverNAT error:no NAT found
^C2025-03-18T00:57:02.785Z      INFO    badger4 v4@v4.3.0/db.go:547     Lifetime L0 stalled for: 0s

2025-03-18T00:57:02.806Z        INFO    badger4 v4@v4.3.0/db.go:626
Level 0 [ ]: NumTables: 00. Size: 0 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 16 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 6 [B]: NumTables: 02. Size: 2.0 MiB of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level Done
```

Tested on:
MacOS Sequoia `15.3.1 (24D70)`
colima `0.8.1` / docker desktop `4.38`
Celestia image `0.21.5` / `0.21.8-mocha`

**Comment by glihm** [comment]
An other variant of the output I experienced without actually changing anything after the node correctly stared, and restarting it:
```
Initializing Celestia Node with command:
celestia light init --p2p.network mocha --rpc.addr=0.0.0.0 --gateway.addr=0.0.0.0
2025-03-18T01:10:11.861Z	WARN	module/gateway	gateway/flags.go:47	custom address or port provided without enabling gateway, setting config values
2025-03-18T01:10:11.861Z	INFO	node	nodebuilder/init.go:31	Initializing Light Node Store over '/home/celestia'
2025-03-18T01:10:11.862Z	INFO	node	nodebuilder/init.go:64	Saved config	{"path": "/home/celestia/config.toml"}
2025-03-18T01:10:11.862Z	INFO	node	nodebuilder/init.go:66	Accessing keyring...
2025-03-18T01:10:11.863Z	WARN	node	nodebuilder/init.go:196	Detected plaintext keyring backend. For elevated security properties, consider using the `file` keyring backend.
2025-03-18T01:10:11.865Z	INFO	node	nodebuilder/init.go:73	Node Store initialized


Starting Celestia Node with command:
celestia light start --core.ip rpc-mocha.pops.one --core.port 9090 --p2p.network mocha

2025-03-18T01:10:11.891Z	INFO	node	nodebuilder/module.go:27	Accessing keyring...
2025-03-18T01:10:11.895Z	INFO	badger4	v4@v4.3.0/levels.go:171	All 22 tables opened in 1ms

2025-03-18T01:10:11.896Z	INFO	badger4	v4@v4.3.0/discard.go:66	Discard stats nextEmptySlot: 13

2025-03-18T01:10:11.896Z	INFO	badger4	v4@v4.3.0/db.go:369	Set nextTxnTs to 60842
2025-03-18T01:10:11.898Z	INFO	module/p2p	p2p/tls.go:54	using a default ws transport
2025/03/18 01:10:11 failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 7168 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Buffer-Sizes for details.
2025-03-18T01:10:11.902Z	INFO	pidstore	pidstore/pidstore.go:67	Loaded peers from disk	{"amount": 98}
2025-03-18T01:10:11.902Z	INFO	module/header	header/config.go:66	No trusted peers in config, initializing with default bootstrappers as trusted peers
2025-03-18T01:10:11.906Z	INFO	header/p2p	p2p/exchange.go:99	client: starting client	{"protocol ID": "/mocha-4/header-ex/v0.0.3"}
2025-03-18T01:10:11.907Z	INFO	pidstore	pidstore/pidstore.go:67	Loaded peers from disk	{"amount": 98}
2025-03-18T01:10:11.907Z	INFO	header/store	store/store.go:193	loaded head	{"height": 291841, "hash": "2BB8FA3F0EE647BEC2B3832DC9492068FC7CA1007722B7392909D8C1D55CB8C6"}
2025-03-18T01:10:11.907Z	INFO	share/discovery	discovery/discovery.go:131	started discovery	{"topic": "/full/v0.1.0"}
2025-03-18T01:10:11.907Z	INFO	share/discovery	discovery/discovery.go:131	started discovery	{"topic": "/archival/v0.1.0"}
2025-03-18T01:10:11.908Z	INFO	fraudserv	fraudserv/helpers.go:29	joining topic	{"id": "/mocha-4/fraud-sub/badencodingv0.1/v0.0.1"}
2025-03-18T01:10:11.908Z	INFO	fraudserv	fraudserv/service.go:114	starting fraud proof service	{"protocol ID": "/mocha-4/fraud/v0.0.1"}
2025-03-18T01:10:12.052Z	INFO	canonical-log	swarm/swarm_dial.go:620	CANONICAL_PEER_STATUS: peer=12D3KooWMsaYHe4QL8XWp1uKrfL3DUnsqFPoMj21j8AysNNuwUTk addr=/ip4/208.115.212.118/udp/2121/quic-v1 sample_rate=100 connection_status="established" dir="outbound"
2025-03-18T01:10:12.367Z	INFO	canonical-log	swarm/swarm_dial.go:620	CANONICAL_PEER_STATUS: peer=12D3KooWAnpFwA6dmSDBzeNQaHD6DMjsLVWDgBnqpXvpKy6CAVMM addr=/ip4/193.34.213.190/udp/2121/quic-v1 sample_rate=100 connection_status="established" dir="outbound"
2025-03-18T01:10:13.753Z	INFO	share/discovery	discovery/discovery.go:333	discovered wanted peers	{"topic": "/archival/v0.1.0", "amount": 5}
2025-03-18T01:10:21.902Z	INFO	basichost	basic/natmgr.go:112	DiscoverNAT error:no NAT found
2025-03-18T01:10:31.908Z	INFO	badger4	v4@v4.3.0/db.go:547	Lifetime L0 stalled for: 0s

2025-03-18T01:10:31.931Z	INFO	badger4	v4@v4.3.0/db.go:626	
Level 0 [ ]: NumTables: 00. Size: 0 B of 0 B. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 16 MiB
Level 1 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 2 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 3 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 4 [ ]: NumTables: 00. Size: 0 B of 10 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 2.0 MiB
Level 5 [B]: NumTables: 05. Size: 8.9 MiB of 10 MiB. Score: 0.00->0.00 StaleData: 18 KiB Target FileSize: 2.0 MiB
Level 6 [ ]: NumTables: 17. Size: 34 MiB of 34 MiB. Score: 0.00->0.00 StaleData: 0 B Target FileSize: 4.0 MiB
Level Done

Error: node: failed to start within timeout(20s): context deadline exceeded
```

**Comment by jcstein** [comment]
could this have something to do with the RPC addr and port in init command @renaynay ?

`celestia light init --p2p.network mocha --rpc.addr=0.0.0.0 --gateway.addr=0.0.0.0`

**Comment by glihm** [comment]
> What version are you on [@glihm](https://github.com/glihm) ?

<img width="332" alt="Image" src="https://github.com/user-attachments/assets/8336e025-25a9-4bec-b7ff-ace51ecd2b0f" />

Those versions, or which tool should I provide the version?

`celestia light init --p2p.network mocha --rpc.addr=0.0.0.0 --gateway.addr=0.0.0.0`

I've seen this line and wondering too. I've tried to setup the container network to `host`, let me try again by tweaking this parameter.

**Comment by jcstein** [comment]
Actually I don't think those should be an issue in terms of connecting to the network, as those are flags for local RPC config of the light node
